### DISCUSSION TITLE
Cookie Consent Popup (GDPR)

### DISCUSSION THREAD

**Discussion by ElCiment** [Budibase/budibase#11937] [Help] [answered]
Hello 

I'm working on a project that needs to be GDPR compliant (Canada), and I'm facing a challenge. I need to add a cookie consent popup to my Budibase application, but I'm not sure how to do it correctly. ( i tried different thing without succes)

I'm reaching out to seek your advice:

Has anyone here successfully added a GDPR-compliant cookie consent popup to their Budibase app?

Are there any plugins or easy-to-follow methods for doing this?

Can you share any tips or code examples that might help?

Your insights will be greatly appreciated, Thank you for your help!

**Comment by ConorWebb96** [comment, marked as answer]
Hey @ElCiment,

This would likely require the creation of a custom component. Here are some examples of the various approved custom [components](https://github.com/budibase/plugins). Here is the documentation on how to create custom [plugins/components](https://docs.budibase.com/docs/custom-plugin).

Just something to note, the created custom component would still need to be added to every page of your app. Additionally, you could create a [feature request](https://github.com/Budibase/budibase/issues/new/choose) on Github if you feel this is something which should be added to the core product.

I hope this helps.
